### PR TITLE
[NAE-1955] Update setData to handle options / choices

### PR DESCRIPTION
**Pull request by Kovy95**
# Description

- add parsing of the options and choices to setData function

Implements [NAE-1955]

### Blocking Pull requests

Depends on #251 

## How Has Been This Tested?

manually, want to add tests

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |     Linux Mint 20      |
| Runtime             |     Java 11      |
| Dependency Manager  |     Maven 3.8.4      |
| Framework version   |    SpringBoot 2.6.2       |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @...
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes:
    - [ ] Lint test
    - [ ] Unit tests
    - [ ] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [ ] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [ ] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [x] I have made corresponding changes to the documentation:
    - [ ] Developer documentation
    - [ ] User Guides
    - [ ] Migration Guides

[NAE-1955]: https://netgrif.atlassian.net/browse/NAE-1955?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Retoocs — APPROVED** [review]

**Review by Retoocs — DISMISSED** [review]

**Review by Retoocs — APPROVED** [review]

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 1]
```diff
@@ -795,8 +817,12 @@ private Object parseFieldsValues(JsonNode jsonNode, DataField dataField) {
                     value = null;
                     break;
                 }
-                String val = node.get("value").asText();
-                value = new I18nString(val);
+                if (node.get("value").isTextual()) {
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:820]: the `isTextual` check may be done in the `parseI18nStringValues` method. This prevents repetition

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 2]
```diff
@@ -919,14 +949,28 @@ private List<String> parseListString(ObjectNode node, String attributeKey) {
...
     }
 
+    private List<I18nString> parseListI18nString(ObjectNode node, String attributeKey) {
+        ArrayNode arrayNode = (ArrayNode) node.get(attributeKey);
+        if (arrayNode == null) {
+            return null;
+        }
+        ArrayList<I18nString> list = new ArrayList<>();
+        arrayNode.forEach(item -> {
+            if (item.isTextual()) {
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:959]: isTextual

**Review by minop — CHANGES_REQUESTED** [review]

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java (hunk 1 above)
Comment (Kovy95) [reply src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:820]: function isnt mine and its also used for other purposes so I think its wont do

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java (hunk 2 above)
Comment (Kovy95) [reply src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:959]: wont do

**Review by tuplle — COMMENTED** [review]
This code is also modified in #251 . Please merge branch NAE-1956 into this.

**Pull request merged**